source/applications/advanced/create_depth_map.py

```diff
@@ -12,57 +13,90 @@
+    Args:
+        frame: Zivid point cloud
 
-    # Getting the point cloud
-    depth_map = frame.point_cloud().copy_data("z")
+    Returns:
+        OpenCV Depth map
 
+    """
+    depth_map = point_cloud.copy_data("z")
     depth_map_uint8 = (
         (depth_map - np.nanmin(depth_map)) / (np.nanmax(depth_map) - np.nanmin(depth_map)) * 255
     ).astype(np.uint8)
 
-    # Applying color map
-    depth_map_color_map = cv2.applyColorMap(depth_map_uint8, cv2.COLORMAP_JET)
+    depth_map_color_map = cv2.applyColorMap(depth_map_uint8, cv2.COLORMAP_VIRIDIS)
 
     # Setting nans to black
     depth_map_color_map[np.isnan(depth_map)[:, :]] = 0
 
-    rgba = frame.point_cloud().copy_data("rgba")
+    return depth_map_color_map
+
+
+def _point_cloud_to_cv_bgr(point_cloud):
+    """Get bgr image from frame.
+
+    Args:
+        frame: Zivid point cloud
+
+    Returns:
+        OpenCV BGR image
+
+    """
+    rgba = point_cloud.copy_data("rgba")
+    # Applying color map
     bgr = cv2.cvtColor(rgba, cv2.COLOR_RGBA2BGR)
 
-    # Displaying the RGB image
-    rgb_window = "ImageRGB"
-    cv2.namedWindow(rgb_window, cv2.WINDOW_NORMAL)
-    cv2.imshow(rgb_window, bgr)
+    return bgr
 
-    # Waiting for the window to be closed
-    print("Close the RGB image to continue")
-    while cv2.getWindowProperty(rgb_window, 0) >= 0:
-        cv2.waitKey(50)
-    cv2.destroyWindow(rgb_window)
 
-    # Saving the RGB image
-    cv2.imwrite(f"{rgb_window}.png", bgr)
+def _visualize_and_save_image(image, image_file, window_name):
+    """Visualize and save image to file.
 
-    # Displaying the Depth map
-    depth_window = "DepthMap"
-    cv2.namedWindow(depth_window, cv2.WINDOW_NORMAL)
-    cv2.imshow(depth_window, depth_map_color_map)
+    Args:
+        image: OpenCV image
+        image_file: File name
+        window_name: OpenCV Window name
 
-    # Waiting for the window to be closed
-    print("Close the Depth map image to continue")
-    while cv2.getWindowProperty(depth_window, 0) >= 0:
+    Returns None
+
+    """
+    cv2.namedWindow(window_name, cv2.WINDOW_NORMAL)
+    cv2.imshow(window_name, image)
+    print("Close the image to continue")
+    while cv2.getWindowProperty(window_name, 0) >= 0:
         cv2.waitKey(50)
-    cv2.destroyWindow(depth_window)
+    cv2.destroyWindow(window_name)
+    cv2.imwrite(image_file, image)
+
+
+def _main():
+
+    app = zivid.Application()
+
+    data_file = Path() / get_sample_data_path() / "Zivid3D.zdf"
+    print(f"Reading ZDF frame from file: {data_file}")
+    frame = zivid.Frame(data_file)
+    point_cloud = frame.point_cloud()
+
+    # TODO: Visualize the point cloud
+
+    print("Converting to BGR image in OpenCV format")
+    bgr = _point_cloud_to_cv_bgr(point_cloud)
+
+    bgr_image_file = "ImageRGB.png"
+    print(f"Visualizing and saving BGR image to file: {bgr_image_file}")
+    _visualize_and_save_image(bgr, bgr_image_file, "BGR image")
+
+    print("Converting to Depth map in OpenCV format")
```

Comment:
Not "OpenCV format", just a Numpy array.

Reply:
done.